sig-and-wg/sig-core/proposals/release-process-improvement.md

```diff
@@ -0,0 +1,107 @@
+# Release process improvements
+
+Created on 2019-01-15 by Krystian Cieslik (@crabtree).
+
+## Status
+
+ Proposed on 2019-01-15
+
+## Current situation:
+
+- Components as source code getting versioned independently from kyma resources folder.
+- Kyma resources folder copied into installer identifies a kyma version (represented by kyma-installer docker image).
+- The release process is taking all the resources and building a kyma version with the lates versions from the components source code.
+
+## Problems:
+
+- The release job consists of two factors - the pipeline definition (described in cont-integration repository) and the pipeline itself defined in Jenkinsfile under the component directory in the consolidated kyma repository. The definition of the pipeline may be removed or renamed as the component evolves. It is hard to determine the particular state of the cont-integration repository from which the release was performed. It may happen that when we remove a component and then remove its job definition the particular bugfix release will no longer be possible. Migration to prow somehow mitigates this problem as the test-infra repository will also have a release branch defined. Unfortunately, jobs configuration is still defined on the master branch of test-infra repository so no job definitions should be deleted until we are sure that we will not need it for a bugfix release.
+- The current master branch version is based on components in a specific version which is not necessarily the latest. Docker images set in the charts not necessarily point to the latest builds of the components. When creating a release branch, suddenly the potential release will consist of all components in the latest version. That is a mismatch which is unexpected and will cause problems. Maybe people by purpose have not updated the component version in the charts. The solution would be either we assure that always the latest version is used (which we tried with the dynamic versioning but people have not accepted it) or we do not change the used versions for a release, which is the current proposal.
+- Each run of release pipeline requires building of all the components - running unit tests, compiling the source code, building and pushing the docker images. Unfortunately, we do not utilize the caching mechanism offered by the docker on our building nodes (jenkins slaves, prow job containers).
```

Comment:
In the last release it took about 12 minutes., and I think we can further optimize it. Is it a real problem?